crates/shielded_token/src/masp/shielded_sync/utils.rs

```diff
@@ -124,6 +148,11 @@ impl Fetched {
     /// block height.
     pub fn contains_height(&self, height: BlockHeight) -> bool {
         self.txs
+            .iter()
+            .map(|(masp_indexed_tx, transaction)| {
+                (masp_indexed_tx.indexed_tx, transaction)
+            })
+            .collect::<BTreeMap<_, _>>()
```

Comment:
Let's not collect a new `BTreeMap` here. The `contains_height` method is supposed to be relatively cheap. We can modify `IndexedTxRange` to refer to a `MaspIndexedTxRange` instead. Lmk if you need help with this.